### PR TITLE
Cg 19 20 complete

### PR DESCRIPTION
**Pull request by cgaddis36**
Added functionality and appropriating tests for:
-Regular user profile Show Page  
-editing user profile attributes
      ~Displaying new attributes on the profile show page 

-Deleted password validation requirement in the user.rb model

### CODE REVIEW

**Review by KellyIB — APPROVED** [review]
Still think it looks good, Chase.

File: app/views/users/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+ Edit Profile
+
+   <%= form_tag "/profile", method: :patch do |f| %>
+
+   <%= text_field_tag :name, "#{@user.name}"%>
+
```
Comment (TravisBorgsmiller) [inline app/views/users/edit.html.erb:7]: You didn't need the @user.name to be string interpolated, I don't think this would work but couldn't tell since the profile edit page doesn't render if you fill out fields incorrectly

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -25,6 +25,21 @@ def show
...
+  def update
+    current_user.update(name: params[:name],
+                        street_address: params[:street_address],
+                        city: params[:city],
+                        state: params[:state],
+                        zip_code: params[:zip_code],
+                        email: params[:email])
+    flash[:notice] = "Your profile has been updated"
+    redirect_to '/profile'
+  end
```
Comment (TravisBorgsmiller) [inline app/controllers/users_controller.rb:42]: We need some kind of conditional here if the user leaves a field as blank or fills out the field incorrectly, this wasn't working functionally in rails s

**Comment by TravisBorgsmiller** [comment]
Would love to also have an edge case test validating that the user entered in correct data and filled out the edit form correctly

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -25,6 +25,21 @@ def show
...
+
+  def update
+    current_user.update(name: params[:name],
+                        street_address: params[:street_address],
+                        city: params[:city],
+                        state: params[:state],
+                        zip_code: params[:zip_code],
+                        email: params[:email])
+    flash[:notice] = "Your profile has been updated"
+    redirect_to '/profile'
```
Comment (TravisBorgsmiller) [inline app/controllers/users_controller.rb]: Also maybe consider adding an exclamation! mark to a flash[:success] message, we want the user to feel excited that he/she updated their profile

**Review by cgaddis36 — COMMENTED** [review]
Thanks for catching the string interpolation in the link_to, sharp eye.  
Good job using the user_params in the update as well.
error_messages fix looks good too

File: app/models/user.rb [hunk 4]
```diff
@@ -1,7 +1,6 @@
 class User < ApplicationRecord
 
   validates_presence_of :name, :street_address, :city, :state, :zip_code
-  validates_presence_of :password, require: true
```
Comment (adumortier) [inline app/models/user.rb:4]: this helped for the update of the password

File: config/routes.rb [hunk 5]
```diff
@@ -45,6 +45,8 @@
   get "/register", to: "users#new"
   post "/users", to: "users#create"
   get "/profile", to: "users#show"
+  get '/profile/edit', to: "users#edit"
+  patch '/profile', to: "users#update"
```
Comment (adumortier) [inline config/routes.rb:49]: let's ask Mike/Meg about the most restful route here

**Review by adumortier — APPROVED** [review]
